controllers/configurationpolicy_controller.go

```diff
@@ -3323,9 +3324,12 @@ func (r *ConfigurationPolicyReconciler) checkAndUpdateResource(
 				diff = handleDiff(log, recordDiff, existingObjectCopy, mergedObjCopy, r.FullDiffs)
 			}
 
+			// Assume object is compliant by inverting the value of throwSpecViolation
 			r.setEvaluatedObject(obj.policy, obj.existingObj, !throwSpecViolation, "")
 
-			return throwSpecViolation, "", diff, updateNeeded, updatedObj
+			matchesAfterDryRun = false
+
+			return throwSpecViolation, "", diff, updateNeeded, updatedObj, matchesAfterDryRun
 		}
 
 		diff = handleDiff(log, recordDiff, existingObjectCopy, dryRunUpdatedObj, r.FullDiffs)
```

Comment:
I feel like we only want to write `properties.matchesAfterDryRun` in some relatively specific cases. So usually we want this function to return false with that, and then the field will be omitted from the status (because false is the "empty" state for booleans).

So I think the `else` case here is when we're in those "relatively specific cases", so this might work:

```suggestion
			return throwSpecViolation, "", diff, updateNeeded, updatedObj, false
		}
		
		matchesAfterDryRun = true

		diff = handleDiff(log, recordDiff, existingObjectCopy, dryRunUpdatedObj, r.FullDiffs)
```

And then all returns lower than this can use
```
		return ..., matchesAfterDryRun
```

Reply:
Originally I was thinking about the bool from the perspective of the policy, as in "return true if the dry run matches the policy assessment". Since the property is in the RelatedObject properties, "return true if the object matches after the dry run update" makes more sense.